### PR TITLE
feat: add option to deploy with enhanced security (closes #34)

### LINKED ISSUE #34: Add docs for enhancing security
Currently this sample is designed to be as easy as possible to work one and deploy.
If deployed in an enterprise context, we usually enforce tighter security rules and we want to explain what can be changed in the sample to improve its security:

#### Tasks
- [x] Create a new docs "Enhance security" in /`docs/enhance-security.md` that explains how to:
    - [ ] ~~Disable exporting keys in output (and also the ability to work locally)~~
    - [x] Configure vnet (and optionally make the app only available on internal network)
    - [ ] ~~Use Azure Key Vault for key and connection string storage~~

### PR DESCRIPTION
**Pull request by sinedied**
## Purpose

- Adds optional VNET with full managed identity support
- Updates packages and infra templates

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type

What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[x] Documentation content changes
[ ] Other... Please describe:
```

### CODE REVIEW

File: docs/enhance-security.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Enhance security
+
+When deploying the sample in a production environment, you may want to enforce tighter security restrictions to protect your data and resources.
+
+One recommendation is to deploy the sample in a [virtual network](https://learn.microsoft.com/azure/virtual-network/virtual-networks-overview) to restrict access to the resources, including the Azure Functions API and the Azure Storage where the documents are stored. In addition, disabling all local authentication methods (ie API keys) and relying on [Azure Entra managed identity](https://learn.microsoft.com/entra/identity/managed-identities-azure-resources/overview) and [RBAC for access control](https://learn.microsoft.com/azure/role-based-access-control/overview) is another way to enhance security.
```
Comment (softchris) [inline docs/enhance-security.md:5]: this sounds like you're just discussing what you can do to enhance security in general, but we're getting both these features below right? e.g `USE_VNET`.

I would slightly reword this to say something like, and make them stand out a bit through bolding them:

To achieve enterprise grade security we've ensured you can leverage the below features through opt-in flags:
- **Deploy in virtual network**, here comes your text
- **Leverage Azure Entra managed identity**, here comes your text

See below how you can opt in to these two features using azd:

wdyt? @sinedied

Comment (sinedied) [reply docs/enhance-security.md:5]: Thanks, I agree this sounds better! I'll update the PR
